src/concrete/ml/onnx/onnx_impl_utils.py

```diff
@@ -265,9 +265,16 @@ def rounded_comparison(
     # Workaround: in this context, `round_bit_pattern` is used as a truncate operation.
     # Consequently, we subtract a term, called `half` that will subsequently be re-added during the
     # `round_bit_pattern` process.
-    half = 1 << (lsbs_to_remove - 1)
+    # half = 1 << (lsbs_to_remove - 1)
 
     # To determine if 'x' 'operation' 'y' (operation being <, >, >=, <=), we evaluate 'x - y'
-    rounded_subtraction = round_bit_pattern((x - y) - half, lsbs_to_remove=lsbs_to_remove)
+    # We cast to int because if half is too high the result might be float
+    # intermediate = ((x - y) - half)
+    # intermediate_as_int = intermediate.astype(numpy.int64)
+    #
+    # if not isinstance(intermediate, Tracer):
+    #     assert (intermediate == intermediate_as_int).all()
+
+    rounded_subtraction = truncate_bit_pattern(x - y, lsbs_to_remove=lsbs_to_remove)
```

Comment:
I see,  I guess I would be more convinced if we test this more than once (like running the associated tests 10 or 100 times) to be sure it does not make them flaky

Reply:
My only concern here is that we merge this truncate feature in the middle of a load from sklearn feature. If we could split this into 2 PR it would be cleaner I think.

Reply:
Discussed offline. Looks like truncate is required for the tests. Good for me to merge as long as we have an appropriate commit message.